src/checkers/inference/model/SourceVariableSlot.java

```diff
@@ -0,0 +1,73 @@
+package checkers.inference.model;
+
+import javax.lang.model.type.TypeMirror;
+
+/**
+ * SourceVariableSlot is a VariableSlot representing a type use in the source code with undetermined value.
+ *
+ * After the Solver is run, each SourceVariableSlot should have an assigned value which is then written
+ * to the output Jaif file for later reinsertion into the original source code.
+ *
+ */
+public class SourceVariableSlot extends VariableSlot {
+
+    /** Actual type wrapped with this TypeMirror. */
```

Comment:
the actual type of the type use

Reply:
refer to value infer

Reply:
**Update**: 
In Jenny's repo, `actualType` is a field of `VariableSlot`, which is incompatible with `ArithmeticVariableSlot`, `RefinementVariableSlot`, etc.

So the `VariableSlot` there is the `SourceVariableSlot` here, and this field should be `SourceVariableSlot`-exclusive.